File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/SpectatorLeft.java
package tech.zolhungaj.amqapi.servercommands.gameroom;

import com.squareup.moshi.Json;
import tech.zolhungaj.amqapi.servercommands.CommandType;

import java.util.Optional;

@CommandType("Spectator Left")
public record SpectatorLeft(
        Optional<Boolean> kicked,
        Optional<String> newHost,
        @Json(name = "spectator") String playerName
){
    public SpectatorLeft{
        if(newHost == null) newHost = Optional.empty();
    }
}

File: src/main/java/tech/zolhungaj/amqapi/servercommands/gameroom/SpectatorLeft.kt
package tech.zolhungaj.amqapi.servercommands.gameroom

import com.squareup.moshi.Json
import tech.zolhungaj.amqapi.servercommands.CommandType

@JvmRecord
@CommandType("Spectator Left")
data class SpectatorLeft(
    val kicked: Boolean,
    val newHost: String?,
    @Json(name = "spectator")
    val playerName: String,
    @Json(name = "readyPingCooldown")
    val readyPingCooldownInSeconds: Int?
)
